rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -211,24 +215,19 @@ RecorderImpl::RecorderImpl(
 RecorderImpl::~RecorderImpl()
 {
   keyboard_handler_->delete_key_press_callback(toggle_paused_key_callback_handle_);
-  if (in_recording_) {
-    stop();
-  }
+  stop();
 }
 
 void RecorderImpl::stop()
 {
-  stop_discovery_ = true;
-  if (discovery_future_.valid()) {
-    auto status = discovery_future_.wait_for(2 * record_options_.topic_polling_interval);
-    if (status != std::future_status::ready) {
-      RCLCPP_ERROR_STREAM(
-        node->get_logger(),
-        "discovery_future_.wait_for(" << record_options_.topic_polling_interval.count() <<
-          ") return status: " << (status == std::future_status::timeout ? "timeout" : "deferred"));
-    }
+  std::lock_guard<std::recursive_mutex> lock(state_transition_mutex_);
```

Comment:
@fujitatomoya, The Clang-Tidy in CLion, giving a warning on this line with `lock.`
```
Previous declaration is here recorder.cpp235:33:
warning: declaration shadows a local variable
```
It would be better to rename `lock` to something like `state_transition_lock`